tox.ini

```diff
@@ -13,7 +13,7 @@ setenv = VIRTUAL_ENV={envdir}
          PYTHONWARNINGS=default::DeprecationWarning
          TESTS_DIR=./esi_leap/tests/
 deps =
-  -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt}
+#  -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt}
```

Comment:
Weirdly, I didn't get any tox errors with this line - what tox command are you running?

I checked a few openstack codebases, and the value they actually use is:

  -c{env:UPPER_CONSTRAINTS_FILE:https://releases.openstack.org/constraints/upper/master}

It may be worth changing ours to that as well?

Reply:
i was just using tox and tox -r in a fresh venv

Reply:
I don't have this issue either.